lib/elastomer_client/client.rb

```diff
@@ -149,6 +148,8 @@ def connection
 
         @connection_block&.call(conn)
 
+        conn.request(:opaque_id) if @opaque_id
```

Comment:
The change looks good, but it is missing a new test. The test should cover creating a client with a middleware that always sets a fake response. The assertion would be that we get the fake response back (instead of an opaque id exception).

The simplest form of such a Faraday middleware is:

```ruby
module Elastomer
  module Middleware
    class MockResponse < Faraday::Middleware
      def initialize(app, &block)
        super(app)
        @response_block = block
      end

      def call(env)
        env.clear_body if env.needs_body?
     
        env.status = 200
        env.response_headers = ::Faraday::Utils::Headers.new
        env.response_headers["Fake"] = "yes"
        env.response = ::Faraday::Response.new

        @response_block&.call(env)

        env.response.finish(env) unless env.parallel?

        env.response
      end
    end
  end
end
```

The middleware above could be used like this:

```ruby
client = ElastomerClient::Client.new(**opts) do |connection|
  connection.request Elastomer::Middleware::MockResponse { |env| env.body = "{}" }
end
```

Reply:
Added a test. Thanks!